Eastern-Research-Group/mywaterway
# Feature/517 glossary cron job

## Related Issues:
* [HMW-517](https://jira.epa.gov/browse/HMW-517)

## Main Changes:
* Created a `cron` task (`node-cron`) that fetches and caches glossary content to the public S3 bucket daily. When not running locally, this task is assigned to the first server instance.
* Updated the client-side glossary components to pull glossary terms from the cached data, rather than the terminology service.

## Steps To Test:
1. Start the application, and verify no error messages are printed to the terminal console.
2. Verify the glossary panel and terms work.
3. In `server.js`, change **line 161** to `'* * * * *',`.
4. Restart the application, wait one minute, then confirm that "Running glossary cron task" is logged.
5. Stop the application.
6. Run the command `npm run glossary` in the app or server directories. Confirm the glossary task completes successfully.

## TODO:
- [ ] Test the S3 configuration after merging.
- [ ] Test that the cron job is only scheduled for one instance after merging.
- [ ] Restore the number of instances in `manifest-dev.yml` after testing the previous two items.
